### PR TITLE
Update the example with the new setup

### PR DESCRIPTION
**Pull request by fvoska**
This PR updates the example to use the new way of setting up DatX with Angular. To sum up the changes:

- init-datx.helper.ts was removed, DatX initialization is now done by `DatxModule` and you can pass it configuration options via `forRoot` method
- `CollectionService` (+tests) and `CollectionTestingService` are now in the lib and do not have to be implemented in the project

I had issues with npm linking and DI tokens so I temporarily created a local build of @datx/jsonapi-angular and installed the tarball just to verify that the changes work.

I will update the branch once we release new version of the lib.

### CODE REVIEW

**Comment by kristijanbambir** [comment]
I don't think datx-jsonapi-angular-next.tgz should be committed to the repository.

File: package.json
Comment (kristijanbambir) [inline package.json]: Instead of pointing to a file, it might be better to point to a repository branch with:
```
npm install "https://github.com/infinum/datx.git#branch-name" --save
```
In any case I would not merge this until datx stable contains these changes.

Comment (fvoska) [reply package.json]: I did this only so that I can test that it works before publishing, I did not intend to merge it like this. I will publish a new version of `@datx/jsonapi-angular` and update the dependency for the example app.

**Review by kristijanbambir — APPROVED** [review]

File: package.json
Comment (fvoska) [reply package.json]: TIL: this type of linking that you suggested does not work because we have a monorepo, so the imports do not work correctly:

```
[0] Error: src/app/services/artists/artists.service.ts:2:35 - error TS2307: Cannot find module '@datx/jsonapi-angular' or its corresponding type declarations.
[0] 
[0] 2 import { CollectionService } from '@datx/jsonapi-angular';
```

This type of link works with source code, but we need the npm artifact with built files.

Anyway, I will update the example after the new version of datx is published.

**Pull request merged**
